### PR TITLE
Hsearch namespace consolidation

### PR DESCRIPTION
**Pull request by jmarchionatto**

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java [hunk 1]
```diff
@@ -1889,9 +1889,9 @@ public void populateFullTextFields(final FhirContext theContext, final IBaseReso
 		} else {
 			theEntity.setNarrativeText(parseNarrativeTextIntoWords(theResource));
 			theEntity.setContentText(parseContentTextIntoWords(theContext, theResource));
-			if (myDaoConfig.isAdvancedLuceneIndexing()) {
-				ExtendedLuceneIndexData luceneIndexData = myFulltextSearchSvc.extractLuceneIndexData(theResource, theNewParams);
-				theEntity.setLuceneIndexData(luceneIndexData);
+			if (myDaoConfig.isAdvancedHSearchIndexing()) {
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java:1892]: I'm not sure about this.  The feature is still called Lucene/Elasticsearch indexing.  We are using the HSearch name for the code.

This is a breaking change in a public api.  At the very least, we need to patch FhirServerConfigCommon in jpserver-starter.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/FulltextSearchSvcImpl.java [hunk 2]
```diff
@@ -182,7 +182,7 @@ private SearchScroll<Long> getSearchScroll(String theResourceType, SearchParamet
 			)
 			.where(
 				f -> f.bool(b -> {
-					ExtendedLuceneClauseBuilder builder = new ExtendedLuceneClauseBuilder(myFhirContext, myModelConfig, b, f);
+					ExtendedHSearchClauseBuilder builder = new ExtendedHSearchClauseBuilder(myFhirContext, myModelConfig, b, f);
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/FulltextSearchSvcImpl.java:185]: Much better name.  👍

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/ObservationLastNIndexPersistSvc.java [hunk 3]
```diff
@@ -103,7 +102,7 @@ private void createOrUpdateIndexedObservation(IBaseResource theResource, Date th
 
 		indexedObservation.setEffectiveDtm(theEffectiveDtm);
 		indexedObservation.setIdentifier(resourcePID);
-		if (myConfig.isStoreResourceInLuceneIndex()) {
+		if (myConfig.isStoreResourceInHSearchIndex()) {
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/ObservationLastNIndexPersistSvc.java:105]: Same problem.  Public api.  Patch jpaserver-starter.

File: hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/term/freetext/ReindexTerminologyHSearchR4Test.java [hunk 4]
```diff
@@ -45,10 +45,10 @@
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 @ExtendWith(SpringExtension.class)
-@ContextConfiguration(classes = {TestR4Config.class, TestHibernateSearchAddInConfig.LuceneFilesystem.class
-	, ReindexTerminologyFreetextR4Test.NoopMandatoryTransactionListener.class
+@ContextConfiguration(classes = {TestR4Config.class, TestHSearchAddInConfig.LuceneFilesystem.class
+	, ReindexTerminologyHSearchR4Test.NoopMandatoryTransactionListener.class
 })
-public class ReindexTerminologyFreetextR4Test extends BaseJpaR4Test {
+public class ReindexTerminologyHSearchR4Test extends BaseJpaR4Test {
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/term/freetext/ReindexTerminologyHSearchR4Test.java:51]: Can we rename the package too?

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/config/DaoConfig.java [hunk 5]
```diff
@@ -272,14 +272,14 @@ public class DaoConfig {
...
-	 * Activates the new Lucene/Elasticsearch indexing of search parameters.
+	 * Activates the new HSearch indexing of search parameters.
 	 * When active, string, token, and reference parameters will be indexed and
 	 * queried within Hibernate Search.
 	 *
 	 * @since 5.6.0
 	 * TODO mb test more with this true
 	 */
-	private boolean myAdvancedLuceneIndexing = false;
+	private boolean myAdvancedHSearchIndexing = false;
```
Comment (michaelabuckley) [inline hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/config/DaoConfig.java:282]: Public api.  At the minimum, this requires a bump of PRE.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/IFulltextSearchSvc.java [hunk 6]
```diff
@@ -67,7 +67,7 @@ public interface IFulltextSearchSvc {
 
 	boolean isDisabled();
 
-	ExtendedLuceneIndexData extractLuceneIndexData(IBaseResource theResource, ResourceIndexedSearchParams theNewParams);
+	ExtendedHSearchIndexData extractLuceneIndexData(IBaseResource theResource, ResourceIndexedSearchParams theNewParams);
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/IFulltextSearchSvc.java:70]: I think this is safe.  I don't think this is called outside Hapi.

**Review by michaelabuckley — APPROVED** [review]
Great.  We need to bump PRE if we're going to change the daoConfig properties, since those public.

File: hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/term/freetext/ReindexTerminologyHSearchR4Test.java (hunk 4 above)
Comment (jmarchionatto) [reply hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/term/freetext/ReindexTerminologyHSearchR4Test.java:51]: Sure. Done

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/config/DaoConfig.java (hunk 5 above)
Comment (jmarchionatto) [reply hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/config/DaoConfig.java:282]: Yes it does.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/ObservationLastNIndexPersistSvc.java (hunk 3 above)
Comment (jmarchionatto) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/ObservationLastNIndexPersistSvc.java:105]: Yes

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java (hunk 1 above)
Comment (jmarchionatto) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java:1892]: Yes. This is code :-)

Doesn't look like breaks starter. It compiles and all tests run well.
For sure changes are required in Smile

**Pull request merged**